src/System.Runtime/tests/System/StringGetHashCodeTests.cs

```diff
@@ -0,0 +1,63 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Xunit;
+
+namespace System.Tests
+{
+    public class StringGetHashCodeTests : RemoteExecutorTestBase
+    {
+        [Fact]
+        /// <summary>
+        /// Ensure that string hash codes are randomized by getting the hash code for the same string in two processes
+        /// and confirming it is different (modulo possible values of int).
+        /// If the legacy hash codes are being returned, it will not be different.
+        /// </summary>
+        public void GetHashCode_UseSameStringInTwoProcesses_ReturnsDifferentHashCodes()
+        {
+            const string abc = "same string in different processes";
+            int childHashCode, parentHashCode = abc.GetHashCode(), timesTried = 0;
+
+            Func<int> GetChildHashCode = () =>
+            {
+                using (RemoteInvokeHandle handle = RemoteInvoke(() => abc.GetHashCode(), new RemoteInvokeOptions { CheckExitCode = false }))
```

Comment:
You may want to have tests for both case-sensitive and case-insensitive hashing. The implementation for the two are using completely different codepaths.

Reply:
And I believe that uapaot has a bug that the case-insensitive hashing is not randomized: https://github.com/dotnet/corert/issues/2588

Reply:
Only case sensitive gets the collision protection though?
```
            if (this.comparer == EqualityComparer<string>.Default)
            {
                this.comparer = (IEqualityComparer<TKey>)NonRandomizedStringEqualityComparer.Default;
            }
```

Reply:
This code fragment is not related to the tests being added in this PR.

It is internal Dictionary performance optimization that is trying to use faster non-randomized string hashcodes if there are not collisions.

Reply:
@jkotas I added more tests for case sensitive and case insensitive string comparer hash code computation.

Please let me know if those are the kinds of tests you were expecting to see.